libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -197,12 +197,26 @@ bool SyntaxChecker::visit(PlaceholderStatement const&)
 	return true;
 }
 
+bool SyntaxChecker::visit(ContractDefinition const& _contract)
+{
+	if (_contract.contractKind() == ContractDefinition::ContractKind::Interface)
+		m_isInterface = true;
```

Comment:
Please add a check with a single file that contains an interface followed by a regular contract, each with a function without visibility specifier :)

Reply:
@chriseth Hehe. It's now being reset (which is also tested).